src/pspm_sf.m

```diff
@@ -184,14 +185,17 @@
   end
   % 3.3 get and filter data
   [sts_load_data, ~, data] = pspm_load_data(model.datafile{iFile}, model.channel);
-  if sts_load_data < 0, return; end
-  Y{1} = data{1}.data; sr(1) = data{1}.header.sr;
-  model.filter.sr = sr(1);
-  [sts_prepdata, Y{2}, sr(2)] = pspm_prepdata(data{1}.data, model.filter);
+  if sts_load_data < 1, return; end
+  y{1} = data{1}.data;
```

Comment:
This is messed up. You use the data from the first channel and the sampling rate from the last channel.

Please change to "data{end}" in line 189, 190. Please replace "data{1}.data" in line 192 with "y{1}".  Please update help text to indicate that *last* data channel is used.

Reply:
Apparently you changed capital Y into small y. That's fine as far as I can see but please check there is no capital Y left in the function.

Reply:
> Apparently you changed capital Y into small y. That's fine as far as I can see but please check there is no capital Y left in the function.

This is to keep the code consistent with `glm` at line 253 (which is what I mentioned on the PsPM meeting last week, but now I see we were not talking about the same thing).

Reply:
> This is messed up. You use the data from the first channel and the sampling rate from the last channel.
> 
> Please change to "data{end}" in line 189, 190. Please replace "data{1}.data" in line 192 with "y{1}". Please update help text to indicate that _last_ data channel is used.

Sure. I think I just need to confirm whether we should leave freedom for users for this as you said *Either the user specifies the marker channel number as a numeric (i.e. the n-th channel out of all channels)* (https://github.com/bachlab/PsPM/pull/493/files#r1237991430)..